### ISSUE TITLE
Command Palette: Add command that lists all possible commands/actions

### ISSUE THREAD

**Issue by liviopv** [WordPress/gutenberg#52380]
## What problem does this address?
It's unclear what the Command Palette can do, and the guidance offered after triggering is minimal ("Type a command or search"). Curiosity goes a long way but it takes several queries to understand what it can or cannot do, which might provide enough friction to its usage.

## What is your proposed solution?
It would be cool to have something similar to the `wp help` command in CLI. Typing 'help' or maybe "command" would provide a curated list of available commands based on the current context, and a link to an external resource documenting all possible commands.

**Comment by annezazu** [comment]
I agree! I think this would be rad to see all listed and make it more intuitive to use. @youknowriad in case this is easy to implement for 6.3.

**Comment by youknowriad** [comment]
There's a lot of commands that are dynamic: depend on the context, depend on what is being typed... So it's not really similar to `wp help`

**Comment by ndiego** [comment]
> There's a lot of commands that are dynamic: depend on the context, depend on what is being typed... So it's not really similar to wp help

Given @youknowriad's comment and the fact that Beta 4 is tomorrow, I don't think this enhancement should be considered for 6.3. @annezazu thoughts?

**Comment by annezazu** [comment]
I agree. Moving out of the board and leaving open for future discussion.

**Comment by richtabor** [comment]
> There's a lot of commands that are dynamic: depend on the context

It could be useful to have a way to see all _available_ commands, within the context you're in. Not sure what the trigger should be to invoke it though.

**Comment by danielbachhuber** [comment]
I just ran into this too. It would be nice to be able to see all registered commands.

**Comment by annezazu** [comment]
Noting that this came up in the [twenty fifth call for testing for the FSE Outreach Program:](https://make.wordpress.org/test/2023/08/09/fse-program-testing-call-25-lets-start-from-the-beginning/#comment-3125)

>I typed help and no results were found. Typing help it could perhaps be helpful to have a doc page show up that one can click and have it open in a new tab. I typed options and nothing showed up. Upon having typed only optio then the Open editor preferences showed up.

**Comment by Mayank-Tripathi32** [comment]
I came across this issue and wanted to work on this proposal for improving command discoverability in the block editor:

Could we register a help command that dynamically displays a list of all currently registered commands, similar to how this?

<img width="1266" height="1096" alt="Image" src="https://github.com/user-attachments/assets/7622f85e-bf1c-4f3c-8534-ceea4f1c2a80" />

As first step, We can retrieve the static list of registered commands using the getCommands selector from: `packages/commands/src/store/selectors.js`

Additionally, I suggest we allow an optional summary field when registering commands. This field could provide a short summary of what each command does, making the help command output more informative.
